### ISSUE TITLE
Windows compile issues

### ISSUE THREAD

**Issue by realityfabric** [RetroShare/RetroShare#201]
I've been trying to link to the header files in order to compile on Windows 10 with Qt Creator.

I got sys/select.h to be included by including the full path (because I could not get Qt Creator to link to the path automatically), but now I'm getting the error

> const sigset_t *__set
> 'sigset_t' does not name a type

on line 35 of sys/select.h

**Comment by electron128** [comment]
If you talk about the #include <sys/select.h> in udplayer.cc which was added by the Haiku guy (34758be52a476616d700225045175d862ddec136), then try to put a #ifndef _WIN32 around it. If it works you can make a pull request. Thanks.

**Comment by defnax** [comment]
i get same error, with that include i commented it out.

**Comment by realityfabric** [comment]
Alright I changed it and it's merged, so I'm closing this

**Comment by realityfabric** [comment]
I submitted the pull request and it's been merged #203

**Comment by realityfabric** [comment]
I tried to use CONFIG+=NO_SQLCIPHER so that only sqlite3.h and not sqlcipher/sqlite3.h was used to compile, but that did not alter which lines the compiler was using, so I edited it like so:

``` C++
//#ifdef NO_SQLCIPHER
#include <sqlite3.h>
//#else
//#include <sqlcipher/sqlite3.h>
//#endif
```

The compiler is still giving me the error `cannot find -lsqlcipher`, whether I comment out the references to that library in the .pro files or not

**Comment by AsamK** [comment]
You need to a full recompile after changing that.

``` bash
make distclean
qmake -r CONFIG+=NO_SQLCIPHER
make
```

**Comment by AsamK** [comment]
Ah sorry, the windows build  config is not set up to use CONFIG+=NO_SQLCIPHER you need to replace LIBS += -lsqlcipher in libretroshare.pro with this:

```
DEFINES *= NO_SQLCIPHER
LIBS += -lsqlite3
```

And then do the recompile

**Comment by realityfabric** [comment]
Alright, it compiled. It says I'm missing some dlls that I'm not missing when I try to run it, but it compiled.

**Comment by realityfabric** [comment]
Ah, it's not actually saying I'm missing the dlls, it's saying the procedure entry points are missing for certain things

> The procedure entry point __cxa_throw_bad_array_new_length could not be located in the dynamic link library C:\Qt\5.5\mingw492_32\bin\Qt5Cored.dll
> 
> The procedure entry point libintl_vasprintf could not be located in the dynamic link library C:\msys64\mingq32\bin\libgnutls-30.dll
> 
> The procedure entry point __gmpn_cnd_add_n could not be located in the dynamic link library C:\msys64\mingw32\bin\libhogweed-4-1.dll

**Comment by hunbernd** [comment]
It seems like, that RS was linked against dll-s that was compiled with different Gcc version and different configurations.

Make sure that these two things are not in your path  at the same time: 
- C:\Qt\5.5\mingw492_32\bin
- C:\msys64\mingq32\bin

For the maximum compatibility it is better to delete both from the path, and copy the files together manually after building.

Check your build configuration in QtCreator, exactly where is the gcc which it is using for compile.
And put only that folder in your path or copy the needed files from there next to the RS executable.

**Comment by cavebeat** [comment]
is this bug fixed?

**Comment by hunbernd** [comment]
There are 3 bugs mixed in this thread, it is hard to follow.
I suggest to close the thread. One can open a new issue if there is still a problem.
- sys/select.h: fixed
- NO_SQLCIPHER: I haven't test it yet, but I look into the files and it seems like it is not fixed
- missing dlls: possible user error, nothing to do
